inlong-audit/audit-service/src/main/java/elector/impl/DBDataSource.java

```diff
@@ -0,0 +1,307 @@
+        int result = 0;
+        try {
+            if ((null == datasource) || (datasource.isClosed())) {
+                initDataSource();
+            }
+
+            Connection connection = datasource.getConnection();
+            try {
+                PreparedStatement pstmt = connection.prepareStatement(sql);
+                try {
+                    result = pstmt.executeUpdate();
+                } catch (Exception executeUpdatEexception) {
+                    logger.error("Exception :{}", executeUpdatEexception.getMessage());
+                } finally {
+                    if (pstmt != null) {
+                        pstmt.close();
+                    }
+                }
+            } catch (Exception pstmtEexception) {
+                logger.error("Exception :{}", pstmtEexception.getMessage());
+            } finally {
+                if (connection != null) {
+                    connection.close();
+                }
+            }
+            getConnectionFailTimes.set(0);
+        } catch (Exception e) {
+            getConnectionFailTimes.addAndGet(1);
+            logger.warn("## get Connection fail ...");
+        }
+        return result;
+    }
+
+    /**
+     * Leader selector
+     */
+    public void leaderSelector() {
+        if (!electorConfig.isUseDefaultLeader()) {
+            try {
+                int result = executeUpdate(electorSql);
+                if (result == 2) {
+                    logger.info(electorConfig.getLeaderId() + " get the leader");
+                } else if (result == 1) {
+                    logger.info(electorConfig.getLeaderId() + " do not get the leader");
+                }
+            } catch (Exception wxception) {
+                logger.error("Exception: {} ,sql:{}", wxception.getMessage(), electorSql);
+            }
+
+        }
+    }
+
+    /**
+     * Replace leader
+     *
+     * @param replaceLeaderId
+     */
+    public void replaceLeader(String replaceLeaderId) {
+        replaceLeaderSql = replaceLeaderSql.replaceAll("#", replaceLeaderId);
+
+        try {
+            int result = executeUpdate(replaceLeaderSql);
+            if (result > 0) {
+                logger.info("## replace leader succ sql:" + replaceLeaderSql);
+            } else {
+                logger.warn("## replace leader fail sql:" + replaceLeaderSql);
+            }
+
+        } catch (Exception exception) {
+            logger.error("Exception :{} ", exception.getMessage());
+        }
+    }
+
+    /**
+     * Release leader
+     */
+    public void releaseLeader() {
+        try {
+            int result = executeUpdate(reLeaseSql);
+            logger.info("ReleaseLeader sql:" + reLeaseSql);
+            if (result == 1) {
+                logger.info(electorConfig.getLeaderId() + " release the leader success");
+            }
+        } catch (Exception exception) {
+            logger.error("Exception {}:,sql:{}", exception.getMessage(), reLeaseSql);
```

Comment:
Print the whole expcetion stack

 logger.error("failed to release leader, sql={}", releaseSql, exception);